traitcuration/static/js/datasources.js

```diff
@@ -9,3 +63,34 @@ function makeDatasourceRequest(selectedDatasource) {
     });
 }
 
+
+// The following functions are customizing progress and error messages for the progress bars in the 'Sources' page
+function progressError(progressBarElement, progressBarMessageElement, excMessage) {
+  progressBarElement.style.backgroundColor = "#B22929";
+  progressBarMessageElement.innerHTML = `Error: ${excMessage}`
+}
+
+
+function olsImportProgress (progressBarElement, progressBarMessageElement, progress){
+  progressBarElement.style.backgroundColor = '#68a9ef';
+  progressBarElement.style.width = progress.percent + "%";
+  progressBarMessageElement.innerHTML = progress.current + ' ontology terms of ' + progress.total + ' processed...';
+}
+
+
+function zoomaImportProgress (progressBarElement, progressBarMessageElement, progress){
+  progressBarElement.style.backgroundColor = '#68a9ef';
+  progressBarElement.style.width = progress.percent + "%";
+  progressBarMessageElement.innerHTML = progress.current + ' traits of ' + progress.total + ' processed...';
+}
+
+function traitImportProgress (progressBarElement, progressBarMessageElement, progress){
+  progressBarElement.style.backgroundColor = '#68a9ef';
+  progressBarElement.style.width = progress.percent + "%";
+  progressBarMessageElement.innerHTML = progress.current + ' MB of ' + progress.total + ' MB downloaded...';
+}
```

Comment:
The blocks for OLS, ZOOMA and ClinVar look very similar (here and also above in the same file). It looks fine for now, but when in the future we add more trait/data sources, we might have to think about how to condense them into a single configurable method.